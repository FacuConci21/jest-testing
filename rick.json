{
    "id": 1,
    "name": "Rick Sanchez",
    "status": "Alive",
    "species": "Human",
    "gender": "Male",
    "universe": "C137",
    "portalGunId": "eafcecca858a7d5fae7f5d7f5ae5fd7",
    "criminalRecord": {
        "sentence": "death",
        "status": "in process",
        "transmitter": "Galactic Federation",
        "universe": "all ..."
    }
}