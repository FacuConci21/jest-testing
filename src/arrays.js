export function arrayFruits() {
    return ['naranja', 'manzana', 'pera', 'banana', 'mandarina'];
}
export function arrayColors() {
    return ['rojo', 'verde', 'azul', 'blanco', 'negro'];
}