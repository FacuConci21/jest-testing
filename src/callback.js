export function theCallbackMuajaja(cb) {
    console.log('hola viajero');
    cb('hola viajero');
}