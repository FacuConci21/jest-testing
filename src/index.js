export function sum(a, b) {
    return a + b;
}
export function subs(a, b) {
    return a - b;
}
export function div(a, b) {
    return a / b;
}
export function prod(a, b) {
    return a * b;
}