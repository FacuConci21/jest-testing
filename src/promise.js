import axios from 'axios';

export function getDataFromAPI(url) {
    return axios.get(url)
        .then(({ data }) => data)
        .finally( () => console.log('done'));
}