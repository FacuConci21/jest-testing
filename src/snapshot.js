export const getCharacter = data => {
    return {
        'id': data.id,
        'name': data.name,
        'status': data.status,
        'species': data.species,
        'gender': data.gender,
        'universe': data.universe,
        "portalGunId": data.portalGunId,
        "criminalRecord": {
            "sentence": data.criminalRecord.sentence,
            "status": data.criminalRecord.status,
            "transmitter": data.criminalRecord.transmitter,
            "universe": data.criminalRecord.universe,
        }
    }
}

/*

Yo no hubiera hecho esto asi, es una sintaxis vieja, hubiera utilizado en su lugar
una Clase de ES6, pero lo dejo de esta manera por seguir el curso simplemente.

*/