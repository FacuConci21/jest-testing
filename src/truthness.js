export function isNull() {
    return null;
}

export function isTrue() {
    return true;
}
export function isFalse() {
    return false;
}
export function isUndefined() {
    return undefined;
}