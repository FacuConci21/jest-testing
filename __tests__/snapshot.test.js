import { getCharacter } from "../src/snapshot";
import rick from "../rick.json";

describe('Es hora de las Instantaneas', () => {
    test('should show a snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot()
    });

    test('should always fails', () => {
        const user = {
            /*  codigo original:
            createAt: Date.now(),
            id: Math.floor(Math.random() * 20),*/
            // datos de la snapshot para que coincida y no moleste.
            createAt: 1611068600352,
            id: 19,
        }
        expect(user).toMatchSnapshot();
    });

    /*test('Instant치nea con algunas Excepciones en sus paths', () => {
        const user = {
          id: Math.floor(Math.random() * 100),
          name: 'Alejandro Gonz치lez Reyes',
          profesion: 'JavaScript and PHP Development',
          createdAt: new Date
        }
    
        // Se espera que la data recibida coincida con la instant치nea. A excepci칩n de los campos id y createdAt, los cuales se espera que reciban cualquier Numero y Fecha respectivamente
        expect(user).toMatchSnapshot({
          id: expect.any(Number),
          createdAt: expect.any(Date)
        });
    });*/
    
    test('We have a exception inside the code', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'tu mama',
        }

        expect(user).toMatchSnapshot({
            id: expect.any(Number),  
        });
    });
    
    
});
