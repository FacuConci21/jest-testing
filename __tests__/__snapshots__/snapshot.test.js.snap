// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las Instantaneas We have a exception inside the code 1`] = `
Object {
  "id": Any<Number>,
  "name": "tu mama",
}
`;

exports[`Es hora de las Instantaneas should always fails 1`] = `
Object {
  "createAt": 1611068600352,
  "id": 19,
}
`;

exports[`Es hora de las Instantaneas should show a snapshot 1`] = `
Object {
  "criminalRecord": Object {
    "sentence": "death",
    "status": "in process",
    "transmitter": "Galactic Federation",
    "universe": "all ...",
  },
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "portalGunId": "eafcecca858a7d5fae7f5d7f5ae5fd7",
  "species": "Human",
  "status": "Alive",
  "universe": "C137",
}
`;
